v2v-helper/migrate/migrate.go

```diff
@@ -283,30 +300,53 @@ func (migobj *Migrate) SyncCBT(ctx context.Context, vminfo vm.VMInfo) error {
 
 			// 11. Copy Changed Blocks over
 			changedBlockCopySuccess := true
+			retries := 0
+			waitTime := 1 * time.Minute
 			migobj.logMessage("Copying changed blocks")
+			for {
+				startTime := time.Now()
+				migobj.logMessage(fmt.Sprintf("Starting incremental block copy for disk %d attempt %d at %s", idx, retries, startTime))
 
-			// incremental block copy
-
-			startTime := time.Now()
-			migobj.logMessage(fmt.Sprintf("Starting incremental block copy for disk %d at %s", idx, startTime))
-
-			err = nbdops[idx].CopyChangedBlocks(ctx, changedAreas, vminfo.VMDisks[idx].Path)
-			if err != nil {
-				select {
-				case <-ctx.Done():
-					err = vmops.CleanUpSnapshots(false)
-					changedBlockCopySuccess = false
-					if err != nil {
-						return errors.Wrap(err, "failed to cleanup snapshot of source VM")
+				err = nbdops[idx].CopyChangedBlocks(ctx, changedAreas, vminfo.VMDisks[idx].Path)
+				if err != nil {
+					select {
+					case <-ctx.Done():
+						err = vmops.CleanUpSnapshots(false)
+						changedBlockCopySuccess = false
+						if err != nil {
+							return errors.Wrap(err, "failed to cleanup snapshot of source VM")
+						}
+					default:
+						if retries >= maxRetries {
+							return errors.Wrap(err, "failed to copy changed blocks, exceeded retries")
+						} else {
+							retries++
+							migobj.logMessage(fmt.Sprintf("Incremental block copy for disk %d attempt %d failed. Retrying in %s", idx, retries, waitTime))
+							// In case if the whole migration is cancelled through context this routine should not be sleeping
+							select {
+							case <-ctx.Done():
```

Comment:
Add comment here, that this Done() happens when there is an error and we are handling it gracefully here.